Meter.c

```diff
@@ -124,26 +136,48 @@ static void BarMeterMode_draw(Meter* this, int x, int y, int w) {
 
    // First draw in the bar[] buffer...
    int offset = 0;
+
+   const Settings* settings = this->host->settings;
+   const wchar_t* barChars = &bars[settings->barType][1];
+   const size_t barLen = wcslen(barChars);
+   const size_t wsub = w * barLen;
+
    for (uint8_t i = 0; i < this->curItems; i++) {
       double value = this->values[i];
+      int actualWidth = 0;
+
       if (isPositive(value) && this->total > 0.0) {
          value = MINIMUM(value, this->total);
+         actualWidth = ceil((value / this->total) * wsub);
          blockSizes[i] = ceil((value / this->total) * w);
       } else {
          blockSizes[i] = 0;
       }
+
+      if ((value / this->total) * wsub < 0.5) {
```

Comment:
Avoid division by zero …
```suggestion
      if (value * wsub < 0.5 * this->total) {
```

Reply:
oh right sorry for that made the same mistake

Reply:
oh wait i think making just one if condition is possible

see the new changes and see if they seem fine

because we need to continue even when total is zero